### ISSUE TITLE
particle keys server uses wrong host address for Gen 3 devices

### ISSUE THREAD

**Issue by avtolstoy** [particle-iot/particle-cli#462]
### Problem

`particle keys server` writes the key with the host set to `$id.udp.particle.io` instead of `$id.udp-mesh.particle.io` when targeting Gen 3 devices, which will divert the devices to non-mesh Device Service instances, preventing mesh-specific CoAP requests from working correctly, affecting both normal operation and setup.

Originally reported in https://community.particle.io/t/recover-from-cli-command-particle-keys-server/46274

**Comment by monkbroc** [comment]
The default key for Gen 3 devices is http://server-public-key.udp-mesh.particle.io

**Comment by avtolstoy** [comment]
```
$ particle --version
1.36.0
$ dfu-util -d 2b04:d00e -a 1 -U server.der -s 3490:128
dfu-util 0.8

Copyright 2005-2009 Weston Schmidt, Harald Welte and OpenMoko Inc.
Copyright 2010-2014 Tormod Volden and Stefan Schmidt
This program is Free Software and has ABSOLUTELY NO WARRANTY
Please report bugs to dfu-util@lists.gnumonks.org

Opening DFU capable USB device...
ID 2b04:d00e
Run-time device DFU version 011a
Claiming USB DFU Interface...
Setting Alternate Setting #1 ...
Determining device status: state = dfuIDLE, status = 0
dfuIDLE, continuing
DFU mode device DFU version 011a
Device returned transfer size 4096
DfuSe interface name: "DCD Flash   "
Upload	[=========================] 100%          128 bytes
Upload done.
$ hexdump -C server.der
00000000  01 18 24 69 64 2e 75 64  70 2d 6d 65 73 68 2e 70  |..$id.udp-mesh.p|
00000010  61 72 74 69 63 6c 65 2e  69 6f 00 ff ff ff ff ff  |article.io......|
00000020  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
*
00000080
$ particle keys server
Okay!  New keys in place, your device will not restart.
$ $ dfu-util -d 2b04:d00e -a 1 -U server1.der -s 3490:128
dfu-util 0.8

Copyright 2005-2009 Weston Schmidt, Harald Welte and OpenMoko Inc.
Copyright 2010-2014 Tormod Volden and Stefan Schmidt
This program is Free Software and has ABSOLUTELY NO WARRANTY
Please report bugs to dfu-util@lists.gnumonks.org

Opening DFU capable USB device...
ID 2b04:d00e
Run-time device DFU version 011a
Claiming USB DFU Interface...
Setting Alternate Setting #1 ...
Determining device status: state = dfuIDLE, status = 0
dfuIDLE, continuing
DFU mode device DFU version 011a
Device returned transfer size 4096
DfuSe interface name: "DCD Flash   "
Upload	[=========================] 100%          128 bytes
Upload done.
$ hexdump -C server1.der
00000000  01 13 24 69 64 2e 75 64  70 2e 70 61 72 74 69 63  |..$id.udp.partic|
00000010  6c 65 2e 69 6f ff ff ff  ff ff ff ff ff ff ff ff  |le.io...........|
00000020  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
*
00000080
```

**Comment by monkbroc** [comment]
Fixed by #470
